libmscore/album.cpp

```diff
@@ -0,0 +1,1489 @@
+        if (tag == "name") {
+            reader.readElementText();
+        } else if (tag == "path") {
+            m_fileInfo.setFile(reader.readElementText());
+            album.setIncludeAbsolutePaths(true);
+        } else if (tag == "relativePath") {
+            if (!m_fileInfo.exists()) {
+                QDir dir(album.fileInfo().dir());
+                QString relativePath = reader.readElementText();
+                m_fileInfo.setFile(dir, relativePath);
+            } else {
+                reader.readElementText();
+            }
+        } else if (tag == "enabled") {
+            m_enabled = reader.readBool();
+        } else if (tag == "pauseDuration") {
+            m_pauseDuration = reader.readDouble();
+        } else {
+            reader.skipCurrentElement();
+        }
+    }
+}
+
+//---------------------------------------------------------
+//   writeAlbumItem
+//---------------------------------------------------------
+
+void AlbumItem::writeAlbumItem(XmlWriter& writer)
+{
+    writer.stag("Score");
+    writer.tag("alias", "");
+    if (album.includeAbsolutePaths()) {
+        writer.tag("path", m_fileInfo.absoluteFilePath());
+    }
+    if (!album.exporting()) {
+        QDir dir(album.fileInfo().dir());
+        QString relativePath = dir.relativeFilePath(m_fileInfo.absoluteFilePath());
+        writer.tag("relativePath", relativePath);
+    } else {
+        writer.tag("relativePath", album.exportedScoreFolder() + QDir::separator() + m_score->title() + ".mscx");
+    }
+    writer.tag("enabled", m_enabled);
+    if (auto lb = getSectionBreak()) {
+        m_pauseDuration = lb->pause();
+    }
+    writer.tag("pauseDuration", m_pauseDuration);
+    writer.etag();
+}
+
+//---------------------------------------------------------
+//   duration
+//---------------------------------------------------------
+
+int AlbumItem::duration() const
+{
+    if (!checkReadiness()) {
+        return -1;
+    }
+    return m_duration;
+}
+
+//---------------------------------------------------------
+//   updateDuration
+//---------------------------------------------------------
+
+void AlbumItem::updateDuration()
+{
+    if (!checkReadiness()) {
+        return;
+    }
+    m_duration = m_score->duration();
+    emit durationChanged();
+}
+
+//---------------------------------------------------------
+//   getSectionBreak
+//---------------------------------------------------------
+
+LayoutBreak* AlbumItem::getSectionBreak() const
+{
+    if (!m_score) {
+        return nullptr;
+    }
+    return m_score->lastMeasure()->sectionBreakElement();
+}
+
+//---------------------------------------------------------
+//   checkReadiness
+///     Used to disallow any actions until a Score has been loaded
+//---------------------------------------------------------
+
+bool AlbumItem::checkReadiness() const
+{
+    if (!m_score) {
+        qDebug() << "You need to load a score before you use an AlbumItem." << endl;
```

Comment:
IIRC `qDebug()` adds the linefeed itself, no `endl` needed. Also causes compiler warnings with Qt 5.15